UM-USElab/gradecraft-development
# 3093 attendance assignment type

### Status
READY

### Description
We need to create an assignment type that streamlines attendance taking.

### Todos
- [x] Allow marking existing assignment types as attendance
- [x] Include options dropdown on index table for quick grading, editing, etc.
- [x] Include ability to go back and mass edit existing attendance assignments
- [ ] #3484: Add ability to import attendance assignments from CSV

### Migrations
YES

### Steps to Test or Reproduce
On a new course, an instructor should be able to bulk create attendance events (assignments) via the newly added Attendance link in the staff nav bar. On this setup page, they can choose a date range and select specific days of the week to check for attendance with specific `open_at` and `due_at` times. The ability to add points is also available.

Once properly configured, the instructor can then click on Configure Events where they will be presented with a mass edit type view for each of the events. Events are automatically persisted once the required name field has been populated. Once an event is saved, they can also choose to upload a media image for the assignment.

Once one or more attendance events are saved, any subsequent visits to the Attendance link will bring you to an index page where you can create new individual events and mass edit existing. Quick grade, edit, copy, and delete are also available via the options dropdown.

As of right now, an attendance event is treated the same as an assignment for all intents and purposes.

### Impacted Areas in Application
Assignments, analytics

======================
Closes #3093
